power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/meta_data.hpp

```diff
@@ -36,29 +36,30 @@ template <class T>
 struct get_component_nan;
 
 // ctype string
-template <class T, bool is_enum = std::is_enum_v<T>>
+template <class T>
 struct ctype_t;
 template <>
-struct ctype_t<double, false> {
-    static constexpr const char* value = "double";
+struct ctype_t<double> {
+    static constexpr CType value = CType::c_double;
 };
 template <>
-struct ctype_t<int32_t, false> {
-    static constexpr const char* value = "int32_t";
+struct ctype_t<int32_t> {
+    static constexpr CType value = CType::c_int32;
 };
 template <>
-struct ctype_t<int8_t, false> {
-    static constexpr const char* value = "int8_t";
+struct ctype_t<int8_t> {
+    static constexpr CType value = CType::c_int8;
 };
 
 template <>
-struct ctype_t<RealValue<false>, false> {
-    static constexpr const char* value = "double[3]";
+struct ctype_t<RealValue<false>> {
+    static constexpr CType value = CType::c_double3;
 };
 template <class T>
-struct ctype_t<T, true> : ctype_t<std::underlying_type_t<T>> {};
+requires(std::is_enum_v<T>) struct ctype_t<T> : ctype_t<std::underlying_type_t<T>> {
```

Comment:
i'm not sure but maybe it's possible to solve this without using inheritance by using `using`

Reply:
Inheritance is the standard way of template meta programming.